jyulb/internal/isothermal/jyu_engine.py

```diff
@@ -313,51 +235,31 @@ def iter_lattices(self, names=None):
             names of specific lattices to be iterated over. If names is not
             given, then all lattices will be iterated over.
 
-        Yields
-        -------
-        ABCLattice
-        """
-        yield self._lat
-
-    def iter_meshes(self, names=None):
-        """Iterate over a subset or all of the meshes.
-
-        Parameters
-        ----------
-        names : sequence of str, optional
-            names of specific meshes to be iterated over. If names is not
-            given, then all meshes will be iterated over.
-
-        Yields
+        Returns
         -------
-        ABCMesh
+        A generator of ABCLattice objects
 
         Raises
         ------
-        NotImplementedError
-           always.
+        ValueError
+            if any one of the names is not in SD
         """
-        message = 'JYUEngine does not handle meshes'
-        raise NotImplementedError(message)
-
-    def iter_particles(self, names=None):
-        """Iterate over a subset or all of the particle containers.
+        self._update_dataset_names()
+        if names is None:
+            for name in self.SD.keys():
+                yield self.SD[name]
+        else:
+            for name in names:
+                if name not in self.SD.keys():
+                    message = 'State data does not contain requested item'
+                    raise ValueError(message)
+                yield self.SD[name]
+
+    def _update_dataset_names(self):
```

Comment:
Okay, I see what you mean. Yeah, right now the "SD" is implicitly contained in the engine and accessed with the get/add methods.  The SSB is supposed to propose how the SD should look like.  I am really not clear on what it should be.  Not too certain when they will be deciding on this.

A guess is that an SD is just a class with the current data set methods (so it would be used like `engine.SD.add_dataset(ds)` ).  But really, I don't know  :) so at the moment there is no insight that could direct this current implementation.